src/libraries/System.Net.HttpListener/src/System/Net/WebSockets/HttpListenerWebSocketContext.cs

```diff
@@ -101,7 +101,15 @@ public class HttpListenerWebSocketContext : WebSocketContext
                 else
                 {
                     // AuthenticationSchemes.Digest, AuthenticationSchemes.Negotiate, AuthenticationSchemes.NTLM.
+#if (WINDOWS)
+                    if (user.Identity is WindowsIdentity windowsIdentity)
```

Comment:
Don't we know user.Identity is a WindowsIdentity here?  The .NET Framework implementation just casts:
https://referencesource.microsoft.com/#System/net/System/Net/WebSockets/HttpListenerWebSocketContext.cs,149